llvm/lib/Frontend/HLSL/HLSLRootSignature.cpp

```diff
@@ -187,6 +193,46 @@ MDNode *MetadataBuilder::BuildRootSignature() {
   return MDNode::get(Ctx, GeneratedMetadata);
 }
 
+MDNode *MetadataBuilder::BuildRootFlags(const RootFlags &Flags) {
+  IRBuilder<> Builder(Ctx);
+  return MDNode::get(Ctx, {
+                              MDString::get(Ctx, "RootFlags"),
+                              ConstantAsMetadata::get(
+                                  Builder.getInt32(llvm::to_underlying(Flags))),
+                          });
```

Comment:
This formatting is absurd - I think clang-format is getting very confused here. It might be worth filing a bug that they're doing something very strange with initializer lists as (second? last?) arguments to a function.

In the meantime, maybe it's better to create these arrays explicitly for readability's sake.
```suggestion
  Metadata *MDs[] = {
      MDString::get(Ctx, "RootFlags"),
      ConstantAsMetadata::get(Builder.getInt32(llvm::to_underlying(Flags))),
  };
  return MDNode::get(Ctx, MDs);
```